rocksdb_admin/admin_handler.cpp

```diff
@@ -1045,14 +1056,16 @@ void AdminHandler::async_tm_startMessageIngestion(
     kafka_watcher_map_[db_name] = kafka_watcher;
   }
 
+  int64_t message_count = 0;
   // With kafka_init_blocking_consumer_timeout_ms set to -1, messages from
   // replay_timestamp_ms to the current are synchronously consumed. The
   // calling thread then returns after spawning a new thread to consume
   // live messages.
   kafka_watcher->StartWith(
       replay_timestamp_ms,
-      [](std::shared_ptr<const RdKafka::Message> message,
-          const bool is_replay) {
+      [message_count, db_name, this](
+          std::shared_ptr<const RdKafka::Message> message,
+          const bool is_replay) mutable {
```

Comment:
Do we still need this?

Reply:
Yes, that's the function signature of the lambda required by the kafka watcher:
https://github.com/pinterest/rocksplicator/blob/97eeb18cdd5479742b98121ff6c6b5d26aa6d25a/common/kafka/kafka_watcher.h#L39

Reply:
Is KafkaMessageHandler used anywhere else? If not, maybe we can remove that, doing this in a separated PR is fine

Reply:
Yes, this kafka library is used by the Muse team as well. This flag was added by them, so don't want to remove it.